workspace/document.go

```diff
@@ -0,0 +1,29 @@
+package workspace
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// Document is a file in a directory.
+type Document struct {
+	Root     string
+	Filepath string
+}
+
+// NewDocument creates a document from a filepath.
+func NewDocument(root, file string) Document {
+	return Document{
+		Root:     root,
+		Filepath: file,
```

Comment:
Should this be Filename?

Reply:
It's a path relative to the root. So for clojure exercises it would be `src/anagram.clj`